types/rlp_unmarshal_storage.go

```diff
@@ -67,46 +104,113 @@ func (t *Transaction) UnmarshalStoreRLPFrom(p *fastrlp.Parser, v *fastrlp.Value)
 		return err
 	}
 
-	if len(elems) != 2 {
-		return fmt.Errorf("expected 2 elements")
+	if len(elems) != 2 && len(elems) != 3 {
```

Comment:
What are the two / three elements?

Reply:
If a transaction has a type, then transaction type comes first. But no place to save transaction type for legacy transaction